be/src/exec/vectorized/schema_scan_node.cpp

```diff
@@ -283,4 +288,19 @@ Status SchemaScanNode::set_scan_ranges(const std::vector<TScanRangeParams>& scan
     return Status::OK();
 }
 
+std::vector<std::shared_ptr<pipeline::OperatorFactory>> SchemaScanNode::decompose_to_pipeline(
+        pipeline::PipelineBuilderContext* context) {
+    auto* morsel_queue_factory = context->morsel_queue_factory_of_source_operator(id());
+    size_t dop = morsel_queue_factory->size();
+
+    size_t buffer_capacity = pipeline::ScanOperator::max_buffer_capacity() * dop;
```

Comment:
in non-pipeline engine, the degree of parallelism of schema scan should be 1.

Reply:
> in non-pipeline engine, the degree of parallelism of schema scan should be 1.

fixed…In fact, `morsel_queue_factory->size()` will only return 1 for SchemaScanOperator, but I think it's more appropriate to explicitly set dop to 1.